internal/services/springcloud/spring_cloud_service_resource.go

```diff
@@ -581,6 +603,22 @@ func resourceSpringCloudServiceRead(d *pluginsdk.ResourceData, meta interface{})
 	if utils.ResponseWasNotFound(serviceRegistry.Response) {
 		serviceRegistryEnabled = false
 	}
+
+	defaultBuildServiceEnabled := true
+	if resp.Sku != nil && resp.Sku.Name != nil && *resp.Sku.Name == "E0" {
+		defaultBuildService, err := buildServiceClient.GetBuildService(ctx, id.ResourceGroup, id.SpringName, "default")
+		if err != nil {
+			if !utils.ResponseWasNotFound(defaultBuildService.Response) {
+				return fmt.Errorf("retrieving default build service of %s: %+v", id, err)
+			}
+			defaultBuildServiceEnabled = false
+		}
+		if utils.ResponseWasNotFound(defaultBuildService.Response) {
+			defaultBuildServiceEnabled = false
+		}
+	}
+	d.Set("default_build_service_enabled", defaultBuildServiceEnabled)
```

Comment:
why are we doing this only for the SKU E0? Shouldn't we create this always

Reply:
Build service is only supported in enterprise sku.